core/src/net/sf/openrocket/aerodynamics/BarrowmanCalculator.java

```diff
@@ -515,39 +515,46 @@ private double calculatePressureDrag(FlightConfiguration configuration, FlightCo
 		base = calculateBaseCD(conditions.getMach());
 		
 		total = 0;
-		for (RocketComponent c : configuration.getActiveComponents()) {
+		final InstanceMap imap = configuration.getActiveInstances();
+	    for(Map.Entry<RocketComponent, ArrayList<InstanceContext>> entry: imap.entrySet() ) {
+			final RocketComponent c = entry.getKey();
 			if (!c.isAerodynamic())
 				continue;
-			
-			// Pressure fore drag
-			double cd = calcMap.get(c).calculatePressureDragForce(conditions, stagnation, base,
-					warnings);
-			total += cd;
-			
-			if (map != null) {
-				map.get(c).setPressureCD(cd);
-			}
-			
-			if(c.isCDOverridden()) continue;					
 
-	
-			// Stagnation drag
-			if (c instanceof SymmetricComponent) {
-				SymmetricComponent s = (SymmetricComponent) c;
+			// iterate across component instances
+			final ArrayList<InstanceContext> contextList = entry.getValue();
+			for(InstanceContext context: contextList ) {
 				
-				if (radius < s.getForeRadius()) {
-					double area = Math.PI * (pow2(s.getForeRadius()) - pow2(radius));
-					cd = stagnation * area / conditions.getRefArea();
-					total += cd;
-					if (map != null) {
-						map.get(c).setPressureCD(map.get(c).getPressureCD() + cd);
-					}
+				// Pressure fore drag
+				double cd = calcMap.get(c).calculatePressureDragForce(conditions, stagnation, base,
+																	  warnings);
+				total += cd;
+				
+				if (map != null) {
+					map.get(c).setPressureCD(cd);
 				}
 				
-				radius = s.getAftRadius();
+				if(c.isCDOverridden()) continue;					
```

Comment:
nit: while I'm not a style nut, I find it better to keep styles consistent so I think the continue should be on the next line. I realize this is carry over from indentation change, but since the code is being touched - maybe worthwhile to make it consistent.

Reply:
Yeah, coding styles are all over the map with this project -- I've been trying to keep my changes to just functional stuff, and not change things that are just style.  All the same, I didn't realize what a complete outlier that one was; fixed it.  I think that was the only one in the files I touched?

Reply:
Yeah, that was the only one - and agree to sticking with the functional changes in general. I commented probably a bit much, but I figure a review is a review :-) Anyways, thanks for fixing that!